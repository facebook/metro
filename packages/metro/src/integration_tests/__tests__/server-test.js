/**
 * Copyright (c) 2013-present, Facebook, Inc.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @emails oncall+javascript_foundation
 * @format
 */

'use strict';

const Metro = require('../../..');

const fetch = require('node-fetch');

jest.unmock('cosmiconfig');

jasmine.DEFAULT_TIMEOUT_INTERVAL = 60 * 1000;

it('should create a server', async () => {
  const config = await Metro.loadConfig({
    config: require.resolve('../metro.config.js'),
  });

  const httpServer = await Metro.runServer(config, {
    reporter: {update() {}},
  });

  try {
    const response = await fetch(
      `http://localhost:${
        config.server.port
      }/TestBundle.bundle?platform=ios&dev=false&minify=true`,
    );

    const body = await response.text();

    if (!response.ok) {
      console.error(body);
      throw new Error(
        'Metro server responded with status code: ' + response.status,
      );
    }

    expect(body.replace(/https?:\/\/[^:]+:[0-9]+/g, '')).toMatchSnapshot();
  } finally {
    httpServer.close();
  }
});
