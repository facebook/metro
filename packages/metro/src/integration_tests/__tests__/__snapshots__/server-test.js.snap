// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a server 1`] = `
"var __DEV__=false,__BUNDLE_START_TIME__=this.nativePerformanceNow?nativePerformanceNow():Date.now(),process=this.process||{};process.env=process.env||{};process.env.NODE_ENV=\\"production\\";
!(function(r){'use strict';r.__r=n,r.__d=function(r,i,o){if(null!=e[i])return;e[i]={dependencyMap:o,factory:r,hasError:!1,importedAll:t,importedDefault:t,isInitialized:!1,publicModule:{exports:{}}}},r.__c=o;var e=o(),t={},i={}.hasOwnProperty;function o(){return e='number'==typeof __NUM_MODULES__?Array(0|__NUM_MODULES__):Object.create(null)}function n(r){var t=r,i=e[t];return i&&i.isInitialized?i.publicModule.exports:d(t,i)}function u(r){var i=r;if(e[i]&&e[i].importedDefault!==t)return e[i].importedDefault;var o=n(i),u=o&&o.__esModule?o.default:o;return e[i].importedDefault=u}function l(r){var o=r;if(e[o]&&e[o].importedAll!==t)return e[o].importedAll;var u,l=n(o);if(l&&l.__esModule)u=l;else{if(u={},l)for(var a in l)i.call(l,a)&&(u[a]=l[a]);u.default=l}return e[o].importedAll=u}var a=!1;function d(e,t){if(!a&&r.ErrorUtils){var i;a=!0;try{i=s(e,t)}catch(e){r.ErrorUtils.reportFatalError(e)}return a=!1,i}return s(e,t)}var c=16,f=65535;function p(r){return{segmentId:r>>>c,localId:r&f}}function s(t,i){!i&&r.__defineModule&&(r.__defineModule(t),i=e[t]);var o=r.nativeRequire;if(!i&&o){var a=p(t),d=a.segmentId;o(a.localId,d),i=e[t]}if(!i)throw Error('Requiring unknown module \\"'+t+'\\".');if(i.hasError)throw _(t,i.error);i.isInitialized=!0;var c=i,f=c.factory,s=c.dependencyMap;try{var v=i.publicModule;return f(r,n,u,l,v,v.exports,s),i.factory=void 0,i.dependencyMap=void 0,v.exports}catch(r){throw i.hasError=!0,i.error=r,i.isInitialized=!1,i.publicModule.exports=void 0,r}}function _(r,e){return Error('Requiring module \\"'+r+'\\", which threw an exception: '+e)}n.unpackModuleId=p,n.packModuleId=function(r){return r.segmentId<<c+r.localId}})('undefined'==typeof global?this:global);
__d(function(g,r,i,a,m,e,d){'use strict';var t=r(d[0]),o=r(d[1]),c=r(d[2]);m.exports={Foo:o,Bar:t,TypeScript:c}},0,[1,2,5]);
__d(function(g,r,i,a,m,e,d){'use strict';var t=r(d[0]);m.exports={type:'bar',foo:t.type}},1,[2]);
__d(function(g,r,i,a,m,e,d){'use strict';var t=r(d[0]);m.exports={type:'foo',asset:t}},2,[3]);
__d(function(g,r,i,a,m,e,d){m.exports=r(d[0]).registerAsset({__packager_asset:!0,httpServerLocation:\\"/assets\\",width:8,height:8,scales:[1],hash:\\"77d45c1f7fa73c0f6c444a830dc42f67\\",name:\\"test\\",type:\\"png\\"})},3,[4]);
__d(function(g,r,i,a,m,e,d){'use strict'},4,[]);
__d(function(g,r,i,a,m,e,d){Object.defineProperty(e,\\"__esModule\\",{value:!0}),e.test=e.type=void 0;e.type='TypeScript';e.test=!0},5,[]);
__r(0);
//# sourceMappingURL=/TestBundle.map?platform=ios&dev=false&minify=true"
`;
